probatus/missing_values/imputation.py

```diff
@@ -0,0 +1,384 @@
+                            add_indicator=True,
+                        ),
+                    ),
+                    ("ohe_cat", OneHotEncoder(handle_unknown="ignore")),
+                ]
+            )
+
+            preprocessor = ColumnTransformer(
+                transformers=[
+                    ("num", numeric_transformer, numeric_columns),
+                    ("cat", categorical_transformer, categorical_columns),
+                ],
+                remainder="passthrough",
+            )
+
+            model_pipeline = Pipeline(
+                steps=[("preprocessor", preprocessor), ("classifier", self.clf)]
+            )
+
+            temp_results = self._calculate_results(
+                X, y, clf=model_pipeline, strategy=strategy
+            )
+
+            results.append(temp_results)
+
+        # If model supports missing values by default, then calculate the scores
+        # on raw data without any imputation.
+        if self.model_na_support:
+
+            categorical_transformer = Pipeline(
+                steps=[
+                    ("ohe_cat", OneHotEncoder(handle_unknown="ignore")),
+                ]
+            )
+
+            preprocessor = ColumnTransformer(
+                transformers=[("cat", categorical_transformer, categorical_columns)],
+                remainder="passthrough",
+            )
+
+            model_pipeline = Pipeline(
+                steps=[("preprocessor", preprocessor), ("classifier", self.clf)]
+            )
+
+            temp_results = self._calculate_results(
+                X, y, clf=model_pipeline, strategy="No Imputation"
+            )
+            results.append(temp_results)
+
+        self.report_df = pd.DataFrame(results)
+        self.report_df.sort_values(by=f"{self.scorer.metric_name} score", inplace=True)
+        self.fitted = True
+        return self
+
+    def _calculate_results(self, X, y, clf, strategy):
+        """
+        Method to calculate the results for a particular imputation strategy.
+
+        Args:
+            X (pd.DataFrame):
+                input variables.
+
+            y (pd.Series):
+                target variable.
+
+            clf (binary classifier,sklearn.Pipeline):
+                A binary classification model, that will used to evaluate various imputation strategies.
+
+            strategy(string):
+                Name of the strategy used for imputation.
+
+        Returns:
+
+            temp_df(dict) :  Dictionary containing the results of the evaluation.
+        """
+        imputation_results = cross_val_score(
+            clf, X, y, scoring=self.scorer.scorer, cv=self.cv, n_jobs=self.n_jobs
+        )
+
+        temp_results = {
+            "strategy": strategy,
+            f"{self.scorer.metric_name} score": np.round(
+                np.mean(imputation_results), 3
+            ),
+            f"{self.scorer.metric_name} std": np.round(np.std(imputation_results), 3),
```

Comment:
Another cool thing that you could compute here is the time required to impute. But this could be a separate issue for the future.

Reply:
Indeed, time calculation is something we can add later.